docs/contributing.md

```diff
@@ -64,6 +64,12 @@ your company's interests, do feel free to approve it.
 This project follows [Google's Open Source Community
 Guidelines](https://opensource.google/conduct/).
 
+## Contributing large patches
+
+While we appreciate that you took the time to design/redesign or rework an 
+existing component, we cannot accept it as easily as that. As it requires a
+architecture review from multiple stakeholders, which we do with 
+[Design Docs](design_docs.md), see the [process here](design_docs.md#process). 
```

Comment:
nit: This sounds like it's written for someone who has already made the mistake of sending a patch for a large feature without asking about it first. I think it sounds friendlier to say something like "Before sending a PR for a large change, please ..."

Reply:
rewritten to be friendlier